python/cuml/preprocessing/model_selection.py

```diff
@@ -14,46 +14,42 @@
 #
 
 import cudf
-from typing import Union, Tuple
-from numba import jit
+import cupy as cp
 import numpy as np
 
-
-@jit
-def _shuffle_idx(idx: np.ndarray):
-    """ Shuffle idx in place which will be used as indices to split a
-    dataframe of size len(np.ndarray)
-    """
-    # TODO this is the bottleneck and should be a gpu operation,
-    # when possible replace with the mlprim mentioned in cuml #659
-    np.random.shuffle(idx)
+from cuml.utils.cupy_utils import test_numba_cupy_version_conflict
+from cuml.utils.numba_utils import PatchedNumbaDeviceArray
+from numba import cuda
+from typing import Union
 
 
 def train_test_split(
-    X: cudf.DataFrame,
-    y: Union[str, cudf.Series],
-    train_size: Union[float, int] = 0.8,
+    X,
+    y,
+    test_size: Union[float, int] = None,
+    train_size: Union[float, int] = None,
     shuffle: bool = True,
-    seed: int = None,
-) -> Tuple[cudf.DataFrame, cudf.DataFrame, cudf.DataFrame, cudf.DataFrame]:
+    seed: Union[int, cp.random.RandomState, np.random.RandomState] = None
+):
     """
-    Partitions the data into four collated dataframes, mimicing sklearn's
-    `train_test_split`
+    Partitions device data into four collated objects, mimicking
+    Scikit-learn's `train_test_split`
 
     Parameters
     ----------
-    X : cudf.DataFrame
+    X : cudf.DataFrame or cuda_array_interface compliant device array
         Data to split, has shape (n_samples, n_features)
-    y : str or cudf.Series
+    y : str, cudf.Series or cuda_array_interface compliant device array
         Set of labels for the data, either a series of shape (n_samples) or
-        the string label of a column in X containing the labels
+        the string label of a column in X (if it is a cuDF DataFrame)
+        containing the labels
     train_size : float or int, optional
         If float, represents the proportion [0, 1] of the data
         to be assigned to the training set. If an int, represents the number
         of instances to be assigned to the training set. Defaults to 0.8
     shuffle : bool, optional
         Whether or not to shuffle inputs before splitting
-    seed : int, optional
+    seed : int, CuPy RandomState or NumPy RandomState optional
         If shuffle is true, seeds the generator. Unseeded by default
```

Comment:
Add GIthub issue for supporting the `stratify` argument. We can add "good first issue" label to it.